lib/extras.py

```diff
@@ -437,6 +437,144 @@ def callproc(self, procname, vars=None):
         return LoggingCursor.callproc(self, procname, vars)
 
 
+class ReplicationConnection(_connection):
+    """A connection that uses `ReplicationCursor` automatically."""
+
+    def __init__(self, *args, **kwargs):
+        """Initializes a replication connection, by adding appropriate replication parameter to the provided dsn arguments."""
+
+        if len(args):
+            dsn = args[0]
+
+            # FIXME: could really use parse_dsn here
+
+            if dsn.startswith('postgres://') or dsn.startswith('postgresql://'):
+                # poor man's url parsing
+                if dsn.rfind('?') > 0:
+                    if not dsn.endswith('?'):
+                        dsn += '&'
+                else:
+                    dsn += '?'
+            else:
+                dsn += ' '
+            dsn += 'replication=database'
+            args = [dsn] + list(args[1:])
+        else:
+            dbname = kwargs.get('dbname', None)
+            if dbname is None:
+                kwargs['dbname'] = 'replication'
+
+            if kwargs.get('replication', None) is None:
+                kwargs['replication'] = 'database' if dbname else 'true'
+
+        super(ReplicationConnection, self).__init__(*args, **kwargs)
+
+        # prevent auto-issued BEGIN statements
+        self.autocommit = True
+
+    def cursor(self, *args, **kwargs):
+        kwargs.setdefault('cursor_factory', ReplicationCursor)
+        return super(ReplicationConnection, self).cursor(*args, **kwargs)
+
+
+"""Streamging replication types."""
+REPLICATION_PHYSICAL = 0
+REPLICATION_LOGICAL  = 1
+
+class ReplicationCursor(_cursor):
+    """A cursor used for replication commands."""
+
+    def identify_system(self):
+        """Get information about the cluster status."""
+
+        self.execute("IDENTIFY_SYSTEM")
+        return dict(zip(['systemid', 'timeline', 'xlogpos', 'dbname'],
+                        self.fetchall()[0]))
+
+    def quote_ident(self, ident):
+        # FIXME: use PQescapeIdentifier or psycopg_escape_identifier_easy, somehow
```

Comment:
When this will be released we would have PQescapeIdentifier exposed, so yes, it will happen.